examples/interactions/src/jvmMain/kotlin/ReconnectOnConnectFailExample.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.rsocket.kotlin.*
+import io.rsocket.kotlin.core.*
+import io.rsocket.kotlin.payload.*
+import io.rsocket.kotlin.transport.*
+import io.rsocket.kotlin.transport.local.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+
+@TransportApi
+fun main(): Unit = runBlocking {
+    val server = LocalServer()
+    RSocketServer().bind(server) {
+        RSocketRequestHandler {
+            requestStream {
+                val data = it.data.readText()
+                val metadata = it.metadata?.readText()
+                println("Server received payload: data=$data, metadata=$metadata")
+
+                flow {
+                    repeat(50) { i ->
+                        delay(100)
+                        emit(Payload("Payload: $i", metadata))
+                        println("Server sent: $i")
+                    }
+                }
+            }
+        }
+    }
+
+    //emulate connection establishment error
+    var first = true
+
+    val rSocket = RSocketConnector {
+        //reconnect 10 times with 1 second delay if connection establishment failed
+        reconnectable(10) {
+            delay(1000)
+            println("RECONNECT")
+            true
+        }
```

Comment:
reconnectable(10) { exception ->
            delay(1000)
            println("RECONNECT after " + exception)
            true
        }

Reply:
I wasn't expecting IllegalStateException here

```
RECONNECT after java.lang.IllegalStateException: Connection establishment failed
```

Reply:
why not? what would you expect here?

Reply:
An rsocket error?

Reply:
In that particular example, connection establishment failed, so it pass original exception of it. So, f.e. if it will be real example, and you will try to connection to wrong url or you don't have internet connection, you will receive original exception.
Do we really need to wrap it here to RSocket error?

Reply:
This seems like a fundamental question about the nature of a local transport.  To me I'd like it to act as much as possible like a remote transport but more efficient through shared memory obviously.  So not pass through mutable objects or exceptions with real stacktraces.

What's your thoughts here?

Reply:
So, in general, local client will always connect successfully until local server is active. And only in that case, it will pass original exception of server close. May be that can be improved.
Also, may be example can be changed somehow, as for now, it's emulate error on establishment.